### PR TITLE
Update Subscribe modal details

### LINKED ISSUE #31867: Ensure template has a clear name and description

### PR DESCRIPTION
**Pull request by lezama**
Fixes #31867

## Proposed changes:

- Changes the name of the modal to: **Jetpack Subscribe modal**
- Adds a description: **A subscribe form that pops up when someone visits your site**
- Moves it from the Footers category to Uncategorized

- Refactor, extract constants and make things relative to Jetpack instead of the current theme
- Fixes this bug:  when switching between areas "Headers", "Footers", "Uncategorized" the template is being added again and again to the lists.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Testing instructions:

* Add this filter to your test site: `add_filter('jetpack_subscriptions_modal_enabled', '__return_true', 12 );`
* In your wp-admin go to Appearance -> Editor -> Library
* Check that the template shows up under Uncategorized
* Make sure that switching between areas doesn't add the modal template again and again
* Check title and descriptions
* Make edits to the Modal and make sure they get persisted and that the actual modal changes in the frontside
* Enjoy and Thanks!

### CODE REVIEW

**Comment by lezama** [comment]
> - Refactor, extract constants and make things relative to Jetpack instead of the current theme

@scruffian could you check this part?
I am struggling to make the preview here pickup the edits to the template:

Screenshot 2023-07-13 at 12 47 52

They do show up in the template edit view:

Screenshot 2023-07-13 at 12 52 13


🙏

**Comment by lezama** [comment]
@scruffian [this commit](https://github.com/Automattic/jetpack/pull/31868#:~:text=roll%20back%20to%20theme%20namespace) makes it work again. I guess this means that modifications to template parts are not supposed to be ported when switching themes? 🤔

**Comment by edanzer** [comment]
Just wanted to confirm that I've tested this and it is working as expected for me. Notes: 
   - The first version of this where we tried to associate the template part with Jetpack vs theme did not work for me (but did for others). The template was loading in the backend, but not in the frontend. Even after completely re-setting my jetpack dev and WP environment/db. 
   - The current version where it loads using the theme in the template part slug does work as expected. 
   - Edits to the template part immediately show on the frontend. 
   - Any edits to the template part are lost when you change themes, which is expected based on how templates in the site editor currently work.

File: projects/plugins/jetpack/modules/subscriptions/subscribe-modal/class-jetpack-subscribe-modal.php [hunk 1]
```diff
@@ -98,7 +109,15 @@ public function get_block_template_filter( $block_template, $id, $template_type
...
-			$query_result[] = $this->get_custom_template();
+			if ( is_array( $query_result ) ) {
+				// find the custom template and return early if we have a custom version in the results.
+				foreach ( $query_result as $template ) {
+					if ( $template->id === self::get_block_template_part_id() ) {
+						return $query_result;
+					}
+				}
+			}
+			$query_result[] = $this->get_template();
```
Comment (edanzer) [inline projects/plugins/jetpack/modules/subscriptions/subscribe-modal/class-jetpack-subscribe-modal.php:120]: Out of curiosity, can you give a quick note for why this change was needed? Just a bit hard for me to follow.

**Review by edanzer — APPROVED** [review]

File: projects/plugins/jetpack/modules/subscriptions/subscribe-modal/class-jetpack-subscribe-modal.php (hunk 1 above)
Comment (lezama) [reply projects/plugins/jetpack/modules/subscriptions/subscribe-modal/class-jetpack-subscribe-modal.php:120]: Good call! something I didn't mentioned on the PR description. Without this PR when switching between areas "Headers", "Footers", "Uncategorized" the template is being added again and again to the lists.
